src/Serialization/Fields.jl

```diff
@@ -410,34 +526,25 @@ end
 
 function save_internal(s::SerializerState, c::acb)
     return Dict(
-        :parent => save_type_dispatch(s, parent(c)),
-        :real => save_type_dispatch(s, real(c)),
-        :imag => save_type_dispatch(s, imag(c))
+        :vector => save_internal(s, [real(c), imag(c)])[:vector],
```

Comment:
I am slighty concerned that we kind of hard-code here the requirement of first producing nested dictionaries, and then serializing them at the end. I still hope that we can arrange the code in a way that allows for other serialization formats which are e.g. more suitable for inter process communication (see issue #1507).

Of course we are already constructing a `Dict` in this function! But this wouldn't be too hard to change (see sketch in #1507). But if we now rely on the return value to be dict...

So maybe we can extract part of function which serializes the `Vector` into a new helper which is called from here (say `save_vector` or so...) and then just also call this from `save_internal(s, ::Vector,...)`

Reply:
save_internal for vector now has an include_parents flag